modules/programs/zsh.nix

```diff
@@ -36,8 +36,7 @@ let
 
       path = mkOption {
         type = types.str;
-        default = relToDotDir ".zsh_history";
-        defaultText = ".zsh_history";
+        default = ".zsh_history";
```

Comment:
I added #452 to introduce `home.stateVersion`, which acts just like NixOS's `system.stateVersion`.

With it this line line would instead change to
```nix
{
default =
  if versionAtLeast config.home.stateVersion "19.03"
  then ".zsh_history"
  else relToDotDir ".zsh_history";
}
```

This would be accompanied by a news entry saying something like

> The 'programs.zsh.history.path' default option has changed from ... to ...
>
> The new default is only applied for state version 19.03 and above.

I guess can have this entry conditioned on the user having the zsh module enabled.

Reply:
@uvNikita that looks like the way forward ? Wanna update the PR ? it's a longstanding annoying issue xD